controllers/users.js

```diff
@@ -410,6 +410,31 @@ const updateSelf = async (req, res) => {
       }
     }
 
+    if (req.body.disabled_roles) {
+      let rolesToDisable;
+      const data = req.body.disabled_roles;
+
+      if (user.disabled_roles !== undefined) {
+        rolesToDisable = user.disabled_roles;
+
+        data.forEach((role) => {
+          const roleIndex = rolesToDisable.indexOf(role);
+          if (roleIndex !== -1) {
+            rolesToDisable.splice(roleIndex, 1);
+          } else {
+            rolesToDisable.push(role);
+          }
+        });
+      } else {
+        rolesToDisable = data;
+      }
+
+      const updatedUser = await userQuery.addOrUpdate({ disabled_roles: rolesToDisable }, userId);
+      if (updatedUser) {
+        return res.status(200).send({ message: "Privilege modified successfully!" });
+      }
```

Comment:
Why is this being done here?

Reply:
Can you please clarify a bit more

Reply:
What was this piece of code previously doing? and what is the new feature that you are adding?

Reply:
So this code was basically used to data to user Object they have been registered with github and have an id associated with their object. moreover this controller was also used to update the user profile as well but now we're not allow user to update that, for that we now require users to setup profile services

In the new feature  we are adding function which set up a value in the userObject `disabled_roles`, ["super_user", "member"], both or just one, we also handle the removal and addition of these roles in the array in this code block only